src/Compilers/CSharp/Test/Emit/Emit/DynamicAnalysis/DynamicAnalysisResourceTests.cs

```diff
@@ -244,6 +244,186 @@ public static void Main()
         }
 
         [Fact]
+        public void TestMethodSpansWithAttributes()
+        {
+            string source = @"
+using System;
+using System.Security;
+
+public class C
+{
+    static int x;
+
+    public static void Main()                       // Method 0
+    {
+        Fred();
+    }
+            
+    [Obsolete()]
+    static void Fred()                              // Method 1
+    {
+    }
+
+    static C()                                      // Method 2
+    {
+        x = 12;
+    }
+
+    [Obsolete()]
+    public C()                                      // Method 3
+    {
+    }
+
+    int Wilma
+    {
+        [SecurityCritical]
+        get { return 12; }                          // Method 4 
+    }
+
+    [Obsolete()]
+    int Betty => 13;                                // Method 5
+
+    [SecurityCritical]
+    int Pebbles()                                   // Method 6
+    {
+        return 3;
+    }
+
+    [SecurityCritical]
+    ref int BamBam(ref int x)                       // Method 7
+    {
+        return ref x;
+    }
+
+    [SecurityCritical]                              // Method 8
+    C(int x)
+    {
+    }
+
+    [Obsolete()]
+    public int Barney => 13;                        // Method 9
+
+    [SecurityCritical]
+    public static C operator +(C a, C b)            // Method 10
+    {
+        return a;
+    }
+}
+";
+
+            var c = CreateCompilationWithMscorlib(Parse(source + InstrumentationHelperSource, @"C:\myproject\doc1.cs"));
+            var peImage = c.EmitToArray(EmitOptions.Default.WithInstrument("Test.Flag"));
+
+            var peReader = new PEReader(peImage);
+            var reader = DynamicAnalysisDataReader.TryCreateFromPE(peReader, "<DynamicAnalysisData>");
+
+            VerifyDocuments(reader, reader.Documents,
+                @"'C:\myproject\doc1.cs' A2-A4-F7-A4-E2-11-AB-FF-E1-61-3E-4D-2B-9F-C1-75-B8-2A-40-A2 (SHA1)");
+
+            Assert.Equal(14, reader.Methods.Length);
+
+            VerifySpans(reader, reader.Methods[0],
+                "(8,4)-(11,5)",
+                "(10,8)-(10,15)");
```

Comment:
Ah, sorry for misunderstanding. I don't know if there is an existing mechanism to achieve that. Comments could indicate intent, but that 's not as robust as mechanical verification.